js/app.js

```diff
@@ -197,29 +197,32 @@ function tagIsSelected(blogTag) {
 }
 
 
+const snippetTypes = {
+    "media_embed": { "property": "media_item__text.value", "tag": "div", "format": "html", "classname": "" },
+    "code_snippet": { "property": "code", "tag": "pre", "format": "text", "classname": "" },
+    "aside_note": { "property": "note_text", "tag": "aside", "format": "html", "classname": "aside-note" }
+    };
 function processContentSnippets(blogSummary, data)
 {
     for (var key in data) {
         if (data.hasOwnProperty(key)) {
-            if (data[key].system.type === "media_embed") {
-                var objMediaEmbed = blogSummary.querySelector("[data-codename='" + key + "']");
-                if (objMediaEmbed !== null) {
-                    var elementMediaEmbed = document.createElement('div');
-                    elementMediaEmbed.innerHTML = data[key].elements.media_item__text.value;
-                    objMediaEmbed.parentNode.replaceChild(elementMediaEmbed, objMediaEmbed);
-                }
-            }
-            if (data[key].system.type === "code_snippet") {
-                var objCodeSnippet = blogSummary.querySelector("[data-codename='" + key + "']");
-                if (objCodeSnippet !== null) {
-                    var elementCodeSnippet = document.createElement('pre');
-                    elementCodeSnippet.textContent = data[key].elements.code.value;
-                    objCodeSnippet.parentNode.replaceChild(elementCodeSnippet, objCodeSnippet);
-                }
+            if (snippetTypes.hasOwnProperty(data[key].system.type)) {
```

Comment:
the `in` keyword would be sufficient here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

Reply:
Ah, thanks!